File: hw2/RaceMonitor_Java/test/Messages/DidNotFinishUpdateTest.java
package Messages;

import Racedata.AthleteRaceStatus;
import org.junit.Test;

import java.rmi.server.ExportException;
import java.time.LocalDateTime;

import static org.junit.Assert.*;

public class DidNotFinishUpdateTest {
    @Test
    public void testCreate() throws Exception {
        DidNotFinishUpdate dnfUpdate = (DidNotFinishUpdate) AthleteUpdate.Create("DidNotFinish,45,8/15/2017 10:52:00 AM");
        assertNotNull(dnfUpdate);

        assertEquals(AthleteRaceStatus.DidNotFinish, dnfUpdate.getUpdateType());
        assertEquals(45, dnfUpdate.getBibNumber());
        assertEquals(LocalDateTime.of(2017,8,15, 10, 52, 0), dnfUpdate.getTimestamp());

        try {
            AthleteUpdate msg = AthleteUpdate.Create("DidNotFinish,45");
            fail("Exception expected");
        } catch (Exception e) {}

        try {
            AthleteUpdate msg = AthleteUpdate.Create("DidNotFinish");
            fail("Exception expected");
        } catch (Exception e) {}

        try {
            AthleteUpdate msg = AthleteUpdate.Create("DidNotFinish,45,bad,8/15/2017 10:52:00 AM");
            fail("Exception expected");
        } catch (Exception e) {}
    }

    @Test
    public void testToString() throws Exception {
        DidNotFinishUpdate dnfUpdate = new DidNotFinishUpdate();
        dnfUpdate.setBibNumber(45);
        dnfUpdate.setTimestamp(LocalDateTime.of(2017, 8, 15, 10, 52, 0));
        assertEquals("DidNotFinish,45,8/15/2017 10:52:00 AM", dnfUpdate.toString());
    }

    @Test
    public void testSettersAndGetters() throws Exception {
        DidNotFinishUpdate dnfUpdate = new DidNotFinishUpdate();
        dnfUpdate.setBibNumber(45);
        dnfUpdate.setTimestamp(LocalDateTime.of(2017, 8, 15, 10, 52, 0));

        assertEquals(AthleteRaceStatus.DidNotFinish, dnfUpdate.getUpdateType());
        assertEquals(45, dnfUpdate.getBibNumber());
        assertEquals(LocalDateTime.of(2017,8,15,10, 52, 0), dnfUpdate.getTimestamp());
    }
}

File: hw2/RaceMonitor_CSharp/RaceDataTesting/Messages/DidNotFinishUpdateTest.cs
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;

using RaceData;
using RaceData.Messages;

namespace RaceDataTesting.Messages
{
    [TestClass]
    public class DidNotFinishUpdateTest
    {
        [TestMethod]
        public void DidNotFinishUpdate_Create()
        {
            var dnfUpdate = (DidNotFinishUpdate)AthleteUpdate.Create("DidNotFinish,45,8/15/2017 10:52:00 AM");
            Assert.IsNotNull(dnfUpdate);

            Assert.AreEqual(AthleteRaceStatus.DidNotFinish, dnfUpdate.UpdateType);
            Assert.AreEqual(45, dnfUpdate.BibNumber);
            Assert.AreEqual(new DateTime(2017, 8, 15, 10, 52, 0), dnfUpdate.Timestamp);

            try
            {
                var msg = AthleteUpdate.Create("DidNotFinish,45");
                Assert.Fail($"Exception expected when creating: msg={msg}");
            }
            catch { /* ignore */ }

            try
            {
                var msg = AthleteUpdate.Create("DidNotFinish");
                Assert.Fail($"Exception expected when creating: msg={msg}");
            }
            catch { /* ignore */ }

            try
            {
                var msg = AthleteUpdate.Create("DidNotFinish,45,bad,8/15/2017 10:52:00 AM");
                Assert.Fail($"Exception expected when creating: msg={msg}");
            }
            catch { /* ignore */ }
        }

        [TestMethod]
        public void DidNotFinishUpdate_ToString()
        {
            var dnfUpdate = new DidNotFinishUpdate
            {
                BibNumber = 45,
                Timestamp = new DateTime(2017, 8, 15, 10, 52, 0)
            };
            Assert.AreEqual("DidNotFinish,45,8/15/2017 10:52:00 AM", dnfUpdate.ToString());
        }


        [TestMethod]
        public void DidNotFinishUpdate_GetterAndSetters()
        {
            DidNotFinishUpdate dnfUpdate = new DidNotFinishUpdate
            {
                BibNumber = 45,
                Timestamp = new DateTime(2017, 8, 15, 10, 52, 0)
            };

            Assert.AreEqual(AthleteRaceStatus.DidNotFinish, dnfUpdate.UpdateType);
            Assert.AreEqual(45, dnfUpdate.BibNumber);
            Assert.AreEqual(new DateTime(2017, 8, 15, 10, 52, 0), dnfUpdate.Timestamp);

        }
    }
}
